cli/internal/config/config_file.go

```diff
@@ -10,6 +10,53 @@ import (
 	"github.com/vercel/turbo/cli/internal/turbopath"
 )
 
+// CLIConfigProvider is an interface for providing configuration values from the CLI
+// It can be implemented by either a pflag.FlagSet struct or a turbostate.Args struct.
+type CLIConfigProvider interface {
```

Comment:
I think we can get away with just making a struct that holds config values and have functions to convert both `pflag.FlagSet` and `ParsedArgsFromRust` to this data struct. Since we're already getting rid of the laziness of viper by explicitly setting values, I think we can eagerly fetch the string values and throw errors at time of struct creation.

If we really want to keep the lazy behavior, I see that viper provides a `FlagValue` and `FlagValueSet` interface that you can use to bind flags if you aren't using pflags. I think it would be a lot of copy pasta to implement `FlagValue` for all of these flags though since the interface has 4 methods (I don't think Go has nice macros like Rust)